data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/parser/KafkaOtelJsonTraceParser.java

```diff
@@ -0,0 +1,42 @@
+package org.opensearch.dataprepper.plugins.kafka.parser;
+
+import com.google.protobuf.util.JsonFormat;
+import io.opentelemetry.proto.trace.v1.ResourceSpans;
+import io.opentelemetry.proto.trace.v1.TracesData;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.trace.Span;
+import org.opensearch.dataprepper.plugins.otel.codec.OTelProtoOpensearchCodec;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+
+/**
+ * Parsing Open Telemetry trace json format from kafka topic to Spans.
+ */
+public class KafkaOtelJsonTraceParser extends OTelProtoOpensearchCodec.OTelProtoDecoder {
```

Comment:
We have a OTelLogsInputCodec that can be used anywhere a codec is allowed.

https://github.com/opensearch-project/data-prepper/blob/f136f48335cae5551a2220f44a17efa95800cf4e/data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OTelLogsInputCodec.java#L25

Ideally, we would create a new `OTelTraceInputCodec` in that project and then use that code directly in Kafka. This would ensure that the same code is used for Kafka inputs as well as reading from S3. Plus, it gives us the ability to read from S3.